File: AEDSII-TP/tp1/tp01/TP01Q10 - Arquivo em C/arquivo.cpp
/*
 * Nome : Diego Da Silva Ferreira
 * Matricula : 71 06 78
 *
 * */

#include <stdio.h>
#include <stdlib.h>

int main (void){

	int n , x;
	double aux;
	scanf("%d" , &n);

	FILE* arq = fopen("novoarq.txt" , "wb");

	for(x = 0; x < n; x++ ){
	
		scanf("%lf" , &aux);
		fwrite( &aux , sizeof(double) , 1 , arq);
	}

	fclose(arq);

	arq = fopen("novoarq.txt" , "rw");

	for(x = 1; x <= n; x++){
	
		//mecanismo para ler o arquivo ao contrario
		fseek(arq , -(x  *  sizeof(double)) , SEEK_END );

		fread( &aux , sizeof(double) , 1 , arq );
		printf( "%g\n" , aux );		
	}

	fclose(arq);
}

File: AEDSII-TP/tp1/tp01/TP01Q09 - Arquivo em Java/arquivo.java
/*
 * Nome: Diego Da Silva Ferreira
 * Matricula : 71 06 78
 *
 * */

import java.io.*;

public class arquivo{

	public static void main (String[] args) throws IOException{
	
		int x = MyIO.readInt();
		double aux;

		RandomAccessFile raf = new RandomAccessFile("novoarq.txt" , "rw");

		for(int a = 0; a < x; a++){
		
			raf.writeDouble(MyIO.readDouble());
		}

		raf.close();


		raf = new RandomAccessFile("novoarq.txt" , "rw");

		x--;
		while(x >= 0){
		
			raf.seek(x * 8);
			aux = raf.readDouble();

			/*mecanismo para evitar printar a 
			 *parte fracionaria caso ela não exista
			 */
			if((aux - (int)aux) > 0){
			
				MyIO.println(aux);
			}
			else{
			
				MyIO.println((int)aux);
			}
		
			x--;
		}
		
		raf.close();
	}
}
